tests/udf/test_buffer_declarations.py

```diff
@@ -0,0 +1,336 @@
+import numpy as np
+import pytest
+
+from libertem.exceptions import UDFException
+from libertem.udf.base import UDF
+from libertem.common.buffers import BufferWrapper
+
+
+"""
+Rules for buffer declarations, `get_result_buffers` and `get_results`:
+
+1) All buffers are declared in `get_result_buffers`
+2) If a buffer is only computed in `get_results`, it should be marked via
+   `use='result_only'` so it isn't allocated on workers
+3) If a buffer is only used as intermediary result, it should be marked via `use='private'`
+4) Not including a buffer in `get_results` means it will either be passed on
+   unchanged, or dropped if `use='private'`
+5) It's an error to omit an `use='result_only'` buffer in `get_results`
+6) It's an error to include a `use='private'` buffer in `get_results`
+7) All results are returned from `Context.run_udf` as `BufferWrapper` instances
+8) By default, if `get_results` is not implemented, `use='private'` buffers are dropped,
+   and others are passed through unchanged
```

Comment:
This is a nice and concise summary. Maybe we can include that in the documentation instead of here as a comment?

Reply:
I copied it into the docs. I mainly used it as a guide for writing the test cases, so I'll keep it also in that file.